### PR TITLE
feat: Analytics with PostHog

### PR DESCRIPTION
**Pull request by amanape**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
- Consent form displayed to users who haven't decided yet
- Use PostHog to send some form of basic analytics

NOTE the logic regarding saving and submitting settings is all-over-the place and not the best it could be. I hope to address this in the near future

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=ghcr.io/all-hands-ai/runtime:4b40018-nikolaik   --name openhands-app-4b40018   ghcr.io/all-hands-ai/runtime:4b40018
```

### CODE REVIEW

**Comment by enyst** [comment]
I didn't test it, but from what I see, it's _opt-in_, is that correct? It's better than opt-out, IMO, from some points of view - including EU stuff.

**Comment by amanape** [comment]
@enyst The consent form is going to display to all users who haven't done the selection. It is default checked by default but that doesn't mean they are opt-in until they press the save preferences button.

**Review by xingyaoww — APPROVED** [review]

File: frontend/src/entry.client.tsx [hunk 1]
```diff
@@ -6,13 +6,25 @@
...
 import { hydrateRoot } from "react-dom/client";
 import { Provider } from "react-redux";
+import posthog from "posthog-js";
 import { SocketProvider } from "./context/socket";
 import "./i18n";
 import store from "./store";
 
+function PosthogInit() {
+  React.useEffect(() => {
+    posthog.init("phc_3ESMmY9SgqEAGBB6sMGK5ayYHkeUuknH2vP6FmWH9RA", {
```
Comment (rbren) [inline frontend/src/entry.client.tsx:19]: should probably put this in `config.json`

**Review by rbren — APPROVED** [review]

File: frontend/src/entry.client.tsx (hunk 1 above)
Comment (amanape) [reply frontend/src/entry.client.tsx:19]: Will do in a follow-up

**Comment by regismesquita** [comment]
Thanks for the info, I am trying to prevent it at deployment level to make it available to multiple users and want to ensure that none enable it by accident.

**Comment by rbren** [comment]
@regismesquita it can be a little hard to control the frontend at the deployment level, but adding a flag to [config.json](https://github.com/All-Hands-AI/OpenHands/blob/main/frontend/public/config.json) would be the right way to do it. Happy to accept a PR here!

**Pull request merged**